### PR TITLE
feat(streaming): all stateful executors can try spill when chunk comes

### PR DESCRIPTION
**Pull request by wcy-fdu**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?
Previously we have introduced mem table spill for hash_join, dynamic filter and materialized executor,  which can reduce oom risk. 
Other stateful exeutors, like top_n may also suffer from big mem table, so this pr add spill for all stateful executors.

perf test result: http://metabase.risingwave-cloud.xyz/dashboard/241-nexmark-blackhole-1cn-anti-affinity-rw-avg-source-throughput?start_date=2023-12-06&namespace=

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

**Comment by kwannoel** [comment]
Thanks for the PR. Can we run benchmark on this to make sure no performance degradation?

File: src/stream/src/executor/hash_agg.rs [hunk 1]
```diff
@@ -571,6 +571,15 @@ impl<K: HashKey, S: StateStore> HashAggExecutor<K, S> {
         Ok(())
     }
 
+    async fn try_commit_state_tables(this: &mut ExecutorInner<K, S>) -> StreamExecutorResult<()> {
```
Comment (stdrc) [inline src/stream/src/executor/hash_agg.rs:574]: What about `try_flush_data` to make it consistent with others?

**Review by stdrc — DISMISSED** [review]
LGTM

**Review by stdrc — COMMENTED** [review]
Rest LGTM

**Review by kwannoel — COMMENTED** [review]
LGTM as well. Waiting for results.

**Review by stdrc — APPROVED** [review]

**Review by kwannoel — APPROVED** [review]

**Pull request merged**